backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/PPOrderCandidateService.java

```diff
@@ -368,4 +377,47 @@ private void syncOrderLineWithBOMLine(
 
 		ppOrderCandidateDAO.saveLine(orderCandidateLine);
 	}
+
+	public void recomputeMaturingCandidates()
+	{
+		ppMaturingCandidatesViewRepo.deleteStaleCandidates();
+		ppMaturingCandidatesViewRepo.streamValidCandidates()
+				.forEach(this::createUpdateCandidate);
+	}
+
+	private void createUpdateCandidate(final @NonNull PPMaturingCandidateV ppMaturingCandidatesV)
```

Comment:
what happens if the creation of one candidate fails?

loud thinking: shouldn't we create each candidate in a separate transaction/savepoint, try/catch and log exception and move on?

Reply:
makes sense.